src/Uno.UWP/Media/SpeechRecognition/SpeechRecognizer.Android.cs

```diff
@@ -15,8 +17,15 @@ private void InitializeSpeechRecognizer()
 			_speechRecognizer = Android.Speech.SpeechRecognizer.CreateSpeechRecognizer(Android.App.Application.Context);
 		}
 
-		public IAsyncOperation<SpeechRecognitionResult> RecognizeAsync()
+		private async Task<SpeechRecognitionResult> RecognizeTaskAsync(CancellationToken ct)
 		{
+			var isPermissionGranted = await PermissionsHelper.TryGetPermission(ct, Android.Manifest.Permission.RecordAudio);
+
+			if (!isPermissionGranted)
+			{
+				throw new Exception("The RECORD_AUDIO permission is either not present in your Manifest or was not accepted prior to attempting a speech recognition.");
```

Comment:
Consider `UnauthorizedAccessException`

Reply:
Good point, will do. I taught about this myself, but then I decided not to, because the UWP uses the normal `Exception` for its permissions, but I think it is a better solution to use the `UnauthorizedAccessException`.